core/src/com/unciv/logic/automation/NextTurnAutomation.kt

```diff
@@ -114,6 +114,15 @@ object NextTurnAutomation {
 
     /** allow AI to spend money to purchase city-state friendship, buildings & unit */
     private fun useGold(civInfo: CivilizationInfo) {
+        if (civInfo.getHappiness() > 0) {
+            // Marry all possible city-states
+            for (unique in civInfo.getMatchingUniques("Can spend Gold to annex or puppet a City-State that has been your ally for [] turns.")) {
```

Comment:
No parameter check?

Reply:
The parameter is read when first allying the city-state, and the check is baked into `canBeMarriedBy`